ArticleTemplates/assets/js/modules/quiz.js

```diff
@@ -1,251 +1,454 @@
 /*global window,console,define */
 define([
-    'bean',
-    'bonzo',
-    'modules/$',
     'smoothScroll',
     'modules/ads'
 ], function (
-    bean,
-    bonzo,
-    $,
     smoothScroll,
     Ads
 ) {
     'use strict';
 
     var modules = {
-        quizInit: function(quiz) {
-            var $quiz = $(quiz);
+            init: function (quiz) {
+                modules.numAnswered = 0;
+                modules.questionCount = 0;
+                modules.isPersonalityQuiz = document.querySelector('.quiz__buckets');
+                modules.moveMPU = modules.IsAdBelowQuiz(quiz);
+                
+                if (modules.isPersonalityQuiz) {
+                    quiz.classList.add('personality-quiz');
+                    modules.setupPersonalityQuizBuckets();
+                    modules.setupPersonalityQuizQuestions();
+                    modules.removePersonalityQuizAnswers();
+                    modules.buildResultsPanel(quiz);
+                } else {
+                    quiz.classList.add('news-quiz');
+                    modules.score = 0;
+                    modules.scoreMessages = modules.getScoreMessages();
+                    modules.setupNewsQuizQuestions();
+                    modules.removeNewsQuizAnswers();
+                    modules.buildScoresPanel(quiz);
+                }
+            
+                quiz.classList.add("loaded");
+            },
+
+            setupNewsQuizQuestions: function () {
+                var i,
+                    correctAnswers = modules.getCorrectAnswers(),
+                    question,
+                    questions = document.querySelectorAll('.quiz__question'),
+                    questionObj;
+
+                for (i = 0; i < questions.length; i++) {
+                    question = questions[i];
+                    questionObj = {};
+                    questionObj.elem = question;
+                    modules.wrapQuestion(question);
+                    questionObj.correctAnswer = correctAnswers[i];
+                    modules.setupNewsQuizAnswers(questionObj);
+                    modules.questionCount++;
+                }
+            },
 
-            if (!$quiz.length) {
-                return false;
-            }
+            setupNewsQuizAnswers: function (questionObj) {
+                var i,
+                    answerCode,
+                    answers = questionObj.elem.querySelectorAll('.question__answer');
 
-            // Do we have an MPU and is it below the quiz?
-            var mpu = $('.advert-slot__wrapper').first(),
-                moveMPU = false;
+                for (i = 0; i < answers.length; i++) {
+                    answerCode = String.fromCharCode(65 + i);
```

Comment:
Clever! Maybe a comment here to explain that it's looping through alpha a=65,b=66 etc